server/api/admin/challs/put.js

```diff
@@ -21,7 +21,24 @@ export default {
       type: 'object',
       properties: {
         data: {
-          type: 'object'
+          type: 'object',
+          properties: {
+            files: {
+              type: 'array',
+              items: {
+                type: 'object',
+                properties: {
+                  name: {
+                    type: 'string'
+                  },
+                  url: {
+                    type: 'string'
+                  }
+                },
+                required: ['name', 'url']
+              }
+            }
+          }
```

Comment:
We should mark the `files` property as required.

Reply:
It's a partial update so there are no required properties

Reply:
This endpoint does a patch / merge, `files` must *not* be required to allow for a patch which does not change the files to succeed. We may wish to implement a separate schema validating challenges as they are written to the database to make sure they conform to the TypeScript interface definition.